modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/IgniteStatisticsImpl.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.calcite.schema;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import com.google.common.collect.ImmutableList;
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelReferentialConstraint;
+import org.apache.calcite.schema.Statistic;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.apache.ignite.internal.processors.query.calcite.trait.IgniteDistribution;
+import org.apache.ignite.internal.processors.query.stat.ColumnStatistics;
+import org.apache.ignite.internal.processors.query.stat.ObjectStatisticsImpl;
+
+/** Calcite statistics wrapper. */
+public class IgniteStatisticsImpl implements Statistic {
+    /** Internal statistics implementation. */
+    private final ObjectStatisticsImpl statistics;
+
+    /**
+     * Constructor.
+     *
+     * @param statistics Internal object statistics or {@code null}.
+     */
+    public IgniteStatisticsImpl(ObjectStatisticsImpl statistics) {
+        this.statistics = statistics;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Double getRowCount() {
+        // TBD: default values.
+        long rows = (statistics == null) ? 1000 : statistics.rowCount();
+
+        return (double)rows;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean isKey(ImmutableBitSet cols) {
+        return false; // TODO
+    }
+
+    /** {@inheritDoc} */
+    @Override public List<ImmutableBitSet> getKeys() {
+        return null; // TODO
+    }
+
+    /** {@inheritDoc} */
+    @Override public List<RelReferentialConstraint> getReferentialConstraints() {
+        return ImmutableList.of();
+    }
+
+    /** {@inheritDoc} */
+    @Override public List<RelCollation> getCollations() {
+        return ImmutableList.of(); // The method isn't used
+    }
+
+    /** {@inheritDoc} */
+    @Override public IgniteDistribution getDistribution() {
+        return null;
+    }
+
+    /**
+     * @return Column statistics map.
+     */
+    public Map<String, ColumnStatistics> getColumnsStatistics() {
```

Comment:
do we really need both `getColumnsStatistics()` and `getColumnStatistics(String colName)` methods?